src/main/scala/cromwell/engine/db/slick/SchemaManager.scala

```diff
@@ -1,34 +1,74 @@
 package cromwell.engine.db.slick
 
+import java.sql.Connection
+
 import com.typesafe.config.Config
 import cromwell.engine.db.LiquibaseUtils
+import liquibase.diff.DiffResult
 import slick.dbio.DBIO
 import slick.driver.JdbcProfile
 import lenthall.config.ScalaConfig._
 
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+
 object SchemaManager {
   /**
-    * Updates the schema using the manager specified in the database configuration.
+    * Returns a new schema manager based on the config key "schema.manager". If the config value is "liquibase", this
+    * method returns a new LiquibaseSchemaManager. If the config value is "slick", this method returns a new
+    * SlickSchemaManager. If the config key "schema.manager" is not present, returns a new LiquibaseSchemaManager.
     *
-    * The configuration key is "schemaManager", and may have a value of "liquibase" or "slick". If left blank uses
-    * "liquibase".
-    *
-    * May require that the schema be empty before running, or may support incremental updates.
+    * @param databaseConfig The database configuration information, with a the "schema.manager" key.
+    * @return Schema manager.
+    */
+  def fromConfig(databaseConfig: Config): SchemaManager = {
+    databaseConfig.getStringOr("schema.manager", "liquibase").toLowerCase match {
+      case "liquibase" => new LiquibaseSchemaManager
+      case "slick" => new SlickSchemaManager
+    }
+  }
+
+  /**
+    * Lends a connection to a block of code.
     *
-    * @param databaseConfig The database configuration information, including the "schemaManager" key.
     * @param profile The slick jdbc profile for accessing the database.
-    * @param schema The schema to create.
+    * @param database The database to use for the connection.
+    * @param block The block of code to run over the connection.
     * @tparam Profile The slick jdbc profile for accessing the database.
-    * @return An asynchronous runner for the update, possibly eventually returning an error.
+    * @tparam T The return type of the block.
+    * @return The return value of the block.
     */
-  def updateSchema[Profile <: JdbcProfile](databaseConfig: Config,
-                                           profile: Profile,
-                                           schema: Profile#SchemaDescription): DBIO[Unit] = {
-    val schemaManager = databaseConfig.getStringOr("schema.manager", "liquibase").toLowerCase match {
-      case "liquibase" => new LiquibaseSchemaManager
-      case "slick" => new SlickSchemaManager
+  def withConnection[Profile <: JdbcProfile, T](profile: Profile, database: Profile#Backend#Database)
+                                               (block: Connection => T): T = {
+    /*
+     TODO: Should this withConnection() method have a (implicit?) timeout parameter, that it passes on to Await.result?
```

Comment:
Re: "TODONE?", hopefully this comment is clearer on what might be added in the future, instead of hard coding `Duration.Inf`.